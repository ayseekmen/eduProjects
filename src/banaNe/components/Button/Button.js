import React from 'react';
import { Text, TouchableOpacity, ActivityIndicator, View } from 'react-native';

import styles from './Button.style';

const Button = ({ text, onPress, loading, icon, theme = 'primary' }) => {

    return (
        <TouchableOpacity
            style={styles[theme].container}
            onPress={onPress}
            disabled={loading}>
            {loading ? (
                <ActivityIndicator color="white" />
            ) : (
                <View style={styles[theme].button_container}>
                    <Text style={styles[theme].text}>{text}</Text>
                </View>
            )}
        </TouchableOpacity>
    );
};

export default Button;