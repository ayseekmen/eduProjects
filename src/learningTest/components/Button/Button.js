import React from "react";
import { TouchableOpacity, Text, View } from "react-native";

const Button = () => {
    return(
        <TouchableOpacity>
            <Text>1</Text>
        </TouchableOpacity>
    )
}

export default Button;