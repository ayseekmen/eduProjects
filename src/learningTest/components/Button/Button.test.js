import React from "react";
import { render } from '@testing-library/react-native'
import Button from "./Button";

test('should match with snapshot', () => {
    const comp = render(<Button />);
    expect(comp).toMatchSnapshot();
})

test('check number', () => {
    expect(5 + 6).toBe(9)
})