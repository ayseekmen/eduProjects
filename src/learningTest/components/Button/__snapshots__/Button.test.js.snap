// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    1
  </Text>
</View>
`;
